official/utils/misc/distribution_utils.py

```diff
@@ -23,7 +23,8 @@
 
 def get_distribution_strategy(num_gpus,
                               all_reduce_alg=None,
-                              turn_off_distribution_strategy=False):
+                              turn_off_distribution_strategy=False,
+                              distribute_strategy=None):
```

Comment:
since you're changing all the callers of this anyway, i wonder if we can clean up this method a bit. 
1) turn_off_distribution_strategy and distribute_strategy seem redundant, what if we make `distribute_strategy=None` behave as the off case?
2) all_reduce_alg should probably be the last arg as it's the least used and should not need to be specified most of the time. 
3) distribute_strategy should likely be the first one if we expect people to use it all the time. 
4) num_gpus prob needs to become optional if we are going to add TPUStrategy here in the future

Reply:
Do we still want three behavior?
1. distribution strategy is turned off
2. auto choose one
3. override distribution strategy if `distribution_strategy` is specified.

If we don't need 2), I agree and your suggestion makes it cleaner.

Reply:
hmm. i think maybe we can forego #2 since it makes everything more complicated. 
it would require changing all the callers though so start specifying strategy, but probably not too bad. 
@tfboyd what do you think?

Reply:
How about this and keep in mind I often make things confusing?
1) distribute_strategy=off  (this would turn it off)
2) distribute_strategy default setting in flags is "default" which means we pick, which means mirrored or onedevice
3) distribute_strategy set to something other than default we try to do what the user wants and error if we have no idea what they are asking for.

Reply:
@guptapriya @yuefengz   what do you think ^^^^?

Reply:
+1 to Toby's suggestion.